Modules/EventNetworkModule.cs

```diff
@@ -106,66 +142,175 @@ private bool ModHooks_OnSetPlayerBoolHook(string name, bool orig)
+                SendEvent(NetEvent.DashSlash);
             }
             else if (name == nameof(PlayerData.hasDashSlash))
             {
-                SendItemObtained(NetItem.GreatSlash);
+                SendEvent(NetEvent.GreatSlash);
+            }
+            else if (name 
+                     is nameof(PlayerData.lurienDefeated) 
+                     or nameof(PlayerData.hegemolDefeated) 
+                     or nameof(PlayerData.monomonDefeated))
+            {
+                SendEvent(NetEvent.Dreamer);
+            }
+            else if (name
+                     is nameof(PlayerData.openedCrossroads)
+                     or nameof(PlayerData.openedDeepnest)
+                     or nameof(PlayerData.openedGreenpath)
+                     or nameof(PlayerData.openedRuins1)
+                     or nameof(PlayerData.openedRuins2)
+                     or nameof(PlayerData.openedFungalWastes)
+                     or nameof(PlayerData.openedHiddenStation)
+                     or nameof(PlayerData.openedRoyalGardens)
+                     or nameof(PlayerData.tollBenchCity)
+                     or nameof(PlayerData.tollBenchAbyss)
+                     or nameof(PlayerData.tollBenchQueensGardens)
+                     or nameof(PlayerData.cityLift1))
+            {
+                SendEvent(NetEvent.Toll);
+            }
+            else if (name is nameof(PlayerData.cityBridge1) or nameof(PlayerData.cityBridge2))
+            {
+                SendEvent(NetEvent.LeverHit);
             }
         }
 
         return orig;
     }
+    
+    private void PlayMakerFSM_OnEnable(On.PlayMakerFSM.orig_OnEnable orig, PlayMakerFSM self)
+    {
+        orig(self);
+        
+        // Make sure that the player causing changes is the speedrunner
+        if (!IsModuleUsed || TheHuntIsOn.SaveData.IsHunter)
+        {
+            return;
+        }
+
+        if (self.name.Equals("Inspect Region") && self.Fsm.Name.Equals("Control") && 
+            self.gameObject.scene.name is "Room_Tram" or "Room_Tram_RG")
+        {
+            self.InsertCustomAction("Send Event", () =>
+            {
+                SendEvent(NetEvent.Tram);
+            }, 1);
+        }
+        else if (self.name.Equals("Stag") && self.Fsm.Name.Equals("Stag Control"))
+        {
+            self.InsertCustomAction("Fade", () =>
+            {
+                SendEvent(NetEvent.Stag);
+            }, 0);
+        }
+        else if (self.name.Equals("Toll Gate Machine") && self.Fsm.Name.Equals("Toll Machine"))
+        {
+            if (self.gameObject.scene.name is "Mines_33" or "Fungus1_31")
+            {
+                self.InsertCustomAction("Open Gates", () =>
+                {
+                    SendEvent(NetEvent.Toll);
+                }, 2);
+            }
+        }
+        else if (self.name.Equals("Dreamnail Hit") && self.Fsm.Name.Equals("ghost_npc_dreamnail"))
+        {
+            // self.InsertCustomAction("Send", () =>
```

Comment:
Is it intended that this is commented out? If so, delete it instead.

Reply:
Resolved in b831c5f.